examples/pubsub/tutorial_pubsub_configuration.c

```diff
@@ -0,0 +1,180 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information. */
+
+/* Includes */
+#include <stdio.h>
+#include <signal.h>
+#include <open62541/server.h>
+#include <open62541/server_config_default.h>
+#include <open62541/plugin/pubsub.h>
+
+/* Global variables */
+volatile UA_Boolean g_running = true;
+
+/* Initialization of signal handler */
+void initSigHandler(void);
+
+/* Function to give user information about correct usage */
+void usage_info(void);
+
+/* Add variable nodes */
+void addVariables(UA_Server *server);
+
+/***** MAIN FUNCTION **********************************************************************************/
+int main(int argc, char** argv)
+{
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+    UA_Boolean loadPubSubFromFile = UA_FALSE;
+#endif
+    UA_UInt16 port = 4840;
+
+    /* 1. Check arguments and set name of PubSub configuration file*/
+    switch(argc)
+    {
+        case 2:
+            port = (unsigned short)atoi(argv[1]);
+            break;
+        case 3:
+            port = (unsigned short)atoi(argv[1]);
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+            loadPubSubFromFile = UA_TRUE;
+#endif
+            break;
+
+        default:
+            usage_info(); 
+    }
+
+    /* Initialize Signal Handler for exiting the process */
+    initSigHandler();
+
+    /* 2. Initialize Server */
+    UA_Server *server = UA_Server_new();
+    UA_ServerConfig *config = UA_Server_getConfig(server);
+    UA_ServerConfig_setMinimal(config, port, NULL); /* creates server on default port 4840 */
+
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+    if(loadPubSubFromFile) 
+    {
+        UA_Server_setPubSubConfigFilename(server, argv[2]);
+    }
+#endif
+
+    /* 3. Add variable nodes to the server */
+    addVariables(server);
+
+    printf("Starting server...\n");
+    UA_StatusCode statusCode = UA_Server_run(server, &g_running);
+    if(statusCode != UA_STATUSCODE_GOOD)
+    {
+        fprintf(stderr, "Server stopped. Status code: 0x%x\n", statusCode);
+        return(-1);
+    }
+    UA_Server_delete(server);
+
+    printf("\n");
+    return 0;
+}
+/******************************************************************************************************/
+
+
+/* Signal handler */
+static void stopHandler(int signum)
+{
+    printf("Exiting...\n");
+    g_running = false;
+}
+
+
+/* Initialization of signal handler */
+void initSigHandler(void)
+{
+    signal(SIGINT, stopHandler);
+    signal(SIGTERM, stopHandler);
+}
+
+
+/* Function to give user information about correct usage */
+void usage_info(void)
+{
+    printf("USAGE: ./example_pubsub_configuration [name of UA_Binary_Config_File]\n");
+    printf("Alternatively, Bin-files can be loaded via configuration method calls.\n\n");
+}
```

Comment:
It would be nice to have a common example/tutorial code structure. You can for example take a look into the  server_encryption example which also uses signals and command line arguments.